coremltools/converters/mil/frontend/torch/torch_op_registry.py

```diff
@@ -127,3 +129,23 @@ def func_wrapper(func):
         # decorator called without argument
         return func_wrapper
     return func_wrapper(_func)
+
+
+def is_torch_fx_node_supported(torch_fx_node: torch.fx.Node) -> bool:
```

Comment:
We should clearly comment about what are we checking this utils.
This is what I understand:
1. `torch_fx_node.op` must be `call_function`
2. `torch_fx_node.target.__name__.lower()` starts with `aten.`
3. The suffix of the name (`xxx` in `aten.xxx`) in `_TORCH_OPS_REGISTRY `

What I think might have issues (please correct me, since I don't understand this clear)
1: Is good!
2: 
(1) We are not explicitly checking prefix must starts with `aten.`. For instance, is it possible that the name "not start with aten.", but still a valid op? For instance, is it possible that there exist a raw string of name `add` in the IR, but not `aten.add`? 
(2) I am curious about the different of the torch_fx_node notation vs our exisiting parsing rule: [here](https://github.com/apple/coremltools/blob/main/coremltools/converters/mil/frontend/torch/internal_graph.py#L188), why is the identification now be `.` not `::`?
(3) Say we support `aten.add`, does it imply or ganranteed that `coreaten.add` or `edge.add` (I don't know what exact prefix will be for other IRs) is supported?

3: `return torch_fx_node_target_name in _TORCH_OPS_REGISTRY` will be missing inplace ops, say `aten.add_`. Should we use [this](https://github.com/apple/coremltools/blob/main/coremltools/converters/mil/frontend/torch/torch_op_registry.py#L15) instead? And please add unittest if there is instead this corner case we are missing.

Reply:
## TLDR
Changes:
1. Added `assert torch_fx_node_target_name.startswith("aten")`
2. Added `assert not torch_fx_node_target_name.endswith("_")`

Explanations: IIUC
1. All should start with "aten"
2. There shouldn't be `::`
3. There shouldn't be inplace op

## Details for Aten and ::
First, at least from what I observe, all ops start with `aten`, and `.` is used instead of `::`

Second, according to [cccclai's comment](https://github.com/apple/coremltools/pull/2081#discussion_r1427583009) (and I tried locally), `torch_fx_node.target` is the PyTorch operator as a python callable, so:
1. Python stuff has `.` for namespace, not `::`
1. Such callables are aten operators, so they must prefix with "aten"

Thrid, IIUC "core aten" is a subset of "aten", and "edge" is "aten" with constraints. In another word, they are all aten ops

## Details for Inplace
PyTorch FX (Functional Transformation) is functional, so there shouldn't be Inplace op

Reply:
## Prefix `aten` and Separator `.`
From what I observe, target name may or may not contain prefix `aten.`, and separator is `.` if any:
1. For usual fx node created by `torch.fx.symbolic_trace`, target is a PyTorch function, i.e. no prefix, no separator
2. For executorch exported fx node, target type is `executorch.exir.dialects.edge._ops.EdgeOp`, whose name has format "aten.xx.yy"

So my implementation is "remove prefix aten if present". Do nothing if absent.

IIUC there wouldn't be prefix `coreaten` nor `edge`, since they are all subset of `aten`

## In-Place
Since here we choose to only deal with `call_function` fx node, all ops are functional, i.e. no in-place

This choice is based on our demand: ExecuTorch exports only functional graph